### PR TITLE
frontend: exposing amount localization hook using context

### PR DESCRIPTION
**Pull request by shonsirsha**
useLocalizedPunctuation may get noisy (log/warnings) if native
locale is empty or invalid. This is due to the hook
being used in many places and thus it runs indpendently
of one another.

Using context API, we can expose this globally and every
consumer will get the same instance of useLocalizedPunctuation.
This reduces the calls of useLocalizedPunctuation which results
in fixing the warning noises.

### CODE REVIEW

File: frontends/web/src/contexts/LocalizationContext.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+type DarkModeContextProps = {
+    decimal: string;
+    group: string;
+  }
+
+
+// Create a Context for the localization values
+const LocalizationContext = createContext<DarkModeContextProps>({}as DarkModeContextProps);
+export default LocalizationContext;
```
Comment (thisconnect) [inline frontends/web/src/contexts/LocalizationContext.ts:11]: DarkModeContextProps -> LocalizationContextProps

nit: I think it's better to prefer named exports if possible

File: frontends/web/src/contexts/LocalizationProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export const LocalizationProvider = ({ children }: TProps) => {
+  const { decimal, group } = useLocalizedPunctuation('');
+
+  return (
+    <LocalizationContext.Provider value={{ decimal, group }}>
+      {children}
+    </LocalizationContext.Provider>
+  );
+};
```
Comment (thisconnect) [inline frontends/web/src/contexts/LocalizationProvider.tsx:17]: nit: most of the codebase uses lower-case filenames, but there are already a few files in context that contain uppercase letters... maybe start using lowercase here?

**Review by thisconnect — CHANGES_REQUESTED** [review]

**Comment by shonsirsha** [comment]
Thanks for the quick review! Ready for review now. PTAL @thisconnect

File: frontends/web/src/components/amount/amount.tsx [hunk 3]
```diff
@@ -17,8 +17,8 @@
 import { useContext } from 'react';
 import { AppContext } from '../../contexts/AppContext';
 import { CoinUnit, ConversionUnit } from './../../api/account';
-import { useLocalizedPunctuation } from '../../hooks/localized';
 import style from './amount.module.css';
+import { AmountLocalizationContext } from '../../contexts/amount-localization-context';
```
Comment (thisconnect) [inline frontends/web/src/components/amount/amount.tsx:21]: how about naming it just LocalizationContext and maybe in the future we move some other Intl, stuff in there?

Here are a few other Intl calles we have and potentially could go int there one day: 
- https://github.com/digitalbitbox/bitbox-wallet-app/blob/365dfd450cedf2f4e47845516f40ad47c4cefd51/frontends/web/src/components/headerssync/headerssync.tsx#L50
- https://github.com/digitalbitbox/bitbox-wallet-app/blob/365dfd450cedf2f4e47845516f40ad47c4cefd51/frontends/web/src/routes/buy/components/countryselect.tsx#L82
- https://github.com/digitalbitbox/bitbox-wallet-app/blob/365dfd450cedf2f4e47845516f40ad47c4cefd51/frontends/web/src/components/transactions/transaction.tsx#L64
- https://github.com/digitalbitbox/bitbox-wallet-app/blob/365dfd450cedf2f4e47845516f40ad47c4cefd51/frontends/web/src/i18n/utils.ts#L39

File: frontends/web/src/contexts/localization-provider.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { ReactNode } from 'react';
+import { useLocalizedPunctuation } from '../hooks/localized';
+import { AmountLocalizationContext } from './amount-localization-context';
+
+type TProps = {
+    children: ReactNode;
+  }
+
+export const AmountLocalizationProvider = ({ children }: TProps) => {
+  const { decimal, group } = useLocalizedPunctuation('');
```
Comment (thisconnect) [inline frontends/web/src/contexts/localization-provider.tsx:10]: rightnow it is hardcoded to empty string, but this should `nativeLocale` 
```js
  const { nativeLocale } = useContext(AppContext);
```

File: frontends/web/src/contexts/amount-localization-context.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { createContext } from 'react';
+
+type AmountLocalizationContextProps = {
+    decimal: string;
+    group: string;
+  }
```
Comment (thisconnect) [inline frontends/web/src/contexts/amount-localization-context.ts:6]: some day we should check again if eslint can do type spacing maybe this https://eslint.style/rules/ts/type-annotation-spacing

**Review by thisconnect — CHANGES_REQUESTED** [review]

File: frontends/web/src/contexts/localization-provider.tsx (hunk 4 above)
Comment (shonsirsha) [reply frontends/web/src/contexts/localization-provider.tsx:10]: Oops. Sorry. Good catch.

File: frontends/web/src/components/amount/amount.tsx (hunk 3 above)
Comment (shonsirsha) [reply frontends/web/src/components/amount/amount.tsx:21]: Yeah, not opposed to naming it Localization.

**Comment by shonsirsha** [comment]
Thanks @thisconnect , PTAL

**Comment by shonsirsha** [comment]
Eh seems like some tests are still failing. Will fix these.

File: frontends/web/src/hooks/localized.test.ts [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { renderHook } from '@testing-library/react';
+import { useLocalizedPunctuation } from './localized';
+import { describe, expect, it } from 'vitest';
+
+
+describe('useLocalizedPunctuation', () => {
```
Comment (shonsirsha) [inline frontends/web/src/hooks/localized.test.ts:22]: Created this unit test and discarded `amount.localized.test`.

**Review by shonsirsha — COMMENTED** [review]
@thisconnect , just fixed the tests. PTAL 🙏

**Review by thisconnect — APPROVED** [review]
tested 0ec1372c8762f6baa56877d88ddf3271c99cd5c8 LGTM

since, there is 2 console.warn one `frontends/web/src/hooks/localized.ts` and another  `frontends/web/src/utils/localize.ts` how about remove one of them?

Both log the same warning, maybe remove the console.warn in localize.ts wdyt?

**Pull request merged**